R/dsl.R

```diff
@@ -132,11 +133,62 @@ add_package_checks <- function(...,
       add_code_step(covr::codecov(quiet = FALSE))
   }
 }
+#' @description
+#' `do_pkgdown_site()` checks if a `id_rsa` env variable is set and builds a pkgdown site.
+#'   A deployment can be avoided by setting `build_only = TRUE`.
+#'
+#' @inheritParams step_build_pkgdown
+#' @inheritParams step_push_deploy
+#' @param build_only Build the pkgdown site but do not deploy it.
+#' @rdname DSL
+#' @export
+#' @importFrom magrittr %>%
+do_pkgdown_site <- function(...,
+  build_only = FALSE,
+  repos = getOption("repos"),
+  path = ".", branch = NULL,
+  remote_url = NULL,
+  commit_message = NULL, commit_paths = ".") {
+
+  #' @description
+  #' 1. A [step_install_deps()] in the `"install"` stage, using the
+  #'    `repos` argument.
+  get_stage("install") %>%
+    add_step(
+      step_install_deps(repos = repos)
+    )
+
+  #' @description
+  #' 1. A [step_setup_ssh()] in the `"before_deploy"` to setup the upcoming deployment.
+  get_stage("before_deploy") %>%
+    add_step(
+      step_setup_ssh()
+    )
+
+  #' 1. A [step_build_pkgdown()] in the `"deploy"` stage
+  get_stage("deploy") %>%
+    add_step(
+      step_build_pkgdown(...)
+    )
+
+  if (isTRUE(build_only)) {
+    ci()$cat_with_color("`build_only = TRUE` was set, skipping deployment")
+  } else {
+    #' 1. A [step_push_deploy()] in the `"deploy"` stage. By default, the deploy is done to the gh-pages branch.
+    #'
+    if (ci()$can_push()) {
+      get_stage("deploy") %>%
+        add_step(step_push_deploy(
```

Comment:
If we split in `step_setup_push_deploy()` and `step_do_push_deploy()`, we can support non-orphaned `gh-pages` .